src/main/java/com/yahoo/identity/services/storage/sql/SqlAccount.java

```diff
@@ -73,4 +90,48 @@ public String getDescription() {
         return this.account.getDescription();
     }
 
+    @Override
+    @Nonnull
+    public Instant getBlockUntilTime() {
+        return Instant.ofEpochMilli(this.account.getBlockUntilTs());
+    }
+
+    @Override
+    @Nonnull
+    public int getConsecutiveFails() {
+        return this.account.getConsecutiveFails();
+    }
+
+    @Override
+    @Nonnull
+    public Boolean verify(@Nonnull String password) {
+        Instant blockUntil = this.getBlockUntilTime();
+        int consecutiveFails = this.getConsecutiveFails();
+        long blockTimeLeft = Instant.now().until(blockUntil, SECONDS);
+
+        if (blockTimeLeft > 0) {
+            return false;
+        }
+
+        if (!account.verify(password)) {
+            if (consecutiveFails >= ABUSE_MAX_TRIES) {
+                long blockTime =
+                    (long) (Math.pow(ABUSE_BLOCK_FACTOR, consecutiveFails - ABUSE_MAX_TRIES) * ABUSE_MIN_BLOCK);
+                this.accountUpdate.setConsecutiveFails(consecutiveFails + 1);
```

Comment:
I think the `accountUpdate` should be a new one for each call of `verify`.

Reply:
Fixed.